desc/plotting.py

```diff
@@ -1627,11 +1626,12 @@ def plot_boundary(eq, phi=None, plot_axis=True, ax=None, return_data=False, **kw
             period=(np.inf, 2 * np.pi, 2 * np.pi),
             guess=grid.nodes,
         ),
+        NFP=eq.NFP,
         sort=False,
     )
 
     if colors is None:
-        colors = _get_cmap(cmap, nz - 1)(np.linspace(0, 1, nz - 1))
+        colors = _get_cmap(cmap, nz)(np.linspace(0, 1, nz))
```

Comment:
This change is because of the switch to the cyclic colormap. It will still work fine for other colormaps, it's just that the "end" color won't get used.